src/libtcod/renderer_xterm.c

```diff
@@ -251,16 +372,19 @@ static TCOD_Error xterm_recommended_console_size(
     int* __restrict rows) {
   fprintf(stdout, "\x1b[%i;%iH", INT_MAX, INT_MAX);
   fflush(stdout);
+  g_waiting_for_get_size = true;
```

Comment:
Is `g_waiting_for_get_size` really necessary?

Reply:
Never mind, I found the place where it was used.  I'd consider replacing all of the current console size related global variables with a single global pointer to a volatile area to be filled with the output.

Reply:
I don't really like this, but I couldn't think of a better way to disambiguate those input codes. What would the volatile area look like?

Reply:
I'm not sure. I'm just thinking of something like this:
```cpp
struct TerminalSizeOut {
  int columns;
  int rows;
  Uint32 timestamp;
};

static struct TerminalSizeOut* g_terminal_size_out = NULL;  // Or pointer to SDL_Event?

...
{
  struct TerminalSizeOut terminal_size_out = {0};
  g_terminal_size_out = &terminal_size_out;
  while (...) {
     // wait for terminal_size_out to be filled.
  }
}
```
Which just replaces the several global variables with a single one.

Reply:
I'll give that a try, and same for all the globals I just added for mouse input.

Reply:
Actually, that means that `xterm_handle_input_escape()` would be checking `g_terminal_size_out != NULL` before writing to it. Is that thread safe, or can it be a problem if the timeout happens betwewen the check and the write?

Reply:
It could cause issues if you aren't careful.  I'd assume there will be a data race somewhere as long as locks aren't being used.  Since multiple threads are using these globals you'll at least need a mutex when interacting with a global that could be modified from another thread.  Either that or use atomics.

Reply:
Ok, I used various structs for the globals and added a mutex.